parachain/primitives/core/src/types.rs

```diff
@@ -3,20 +3,14 @@
 use codec::{Decode, Encode};
 use enum_iterator::IntoEnumIterator;
 use frame_support::{scale_info::TypeInfo, RuntimeDebug};
-use sp_core::H256;
+use sp_core::{H160, H256};
 use sp_runtime::DigestItem;
 use sp_std::vec::Vec;
 
 #[derive(Encode, Decode, Copy, Clone, PartialEq, Eq, RuntimeDebug, TypeInfo)]
-pub struct MessageId {
-	pub channel_id: ChannelId,
-	pub nonce: u64,
-}
-
-impl MessageId {
-	pub fn new(channel_id: ChannelId, nonce: u64) -> Self {
-		Self { channel_id, nonce }
-	}
+pub enum MessageId {
+	Basic(ChannelId, H160, u64),
+	Incentivized(ChannelId, u64),
```

Comment:
@vgeddes I'm also not happy with the use of `enum` for `MessageId` yet. The `ChannelId` field is redundant because we already have that in the enum variant. But to remove `ChannelId` we'd need custom `Encode` and `Decode` implementations that include the channel id in the serialized `MessageId`. Is it worth going to that effort? As-is, incorrectly use looks like `MessageId::Basic(ChannelId::Incentivized...`, which seems easy to spot.

Reply:
I didnt see this as a redundancy. The enum represents the channel type where as channel id represents an instance of a channel. Unless we are saying that we will always have one instance of a channel per channel type. i.e. We will never have two different incentivized channels that we use for different purposes?

Reply:
@doubledup moving over to enum's here is the right approach. And you can drop the `ChannelId` field, isn't needed at all since client's can inspect the enum variant using match statements to figure that out.

Also  see my comment further above for details on `MessageId`. The type actually isn't that important.

Reply:
> Do we have any plans for multiple channels of a given type?

Nope, not yet. Though I do have vague ideas about supporting multiple "lanes" within a channel. And each lane would have different delivery properties (slow, fast, etc). But this is all in the future.